x-pack/plugins/maps/public/classes/layers/wizards/solution_layers/observability/create_layer_descriptor.ts

```diff
@@ -152,10 +150,10 @@ export function createLayerDescriptor({
   if (!layer || !metric || !display) {
     return null;
   }
-
   const apmSourceQuery = createAmpSourceQuery(layer);
   const label = createLayerLabel(layer, metric);
   const metricsDescriptor = createAggDescriptor(metric);
+  const apmDataViewId = getDataViewId(getSpaceId());
```

Comment:
This view relies on the APM static data view being created and saved. If this data view is not available or has been deleted, it can lead to bugs. 
 For that reason I would try to avoid using the APM static data views in other plugins as much as possible and use the ad hoc data views instead. 
 
 What do you think?

Reply:
That's fair, let me think about this a bit. I'll see how the ad-hoc data view could fit in this case. I'll leave the static solution as a fallback for now.

Reply:
@kpatticha I spoke to @nreese and he explained that an ad-hoc data view won't work for' maps'. Maps stores the data view id in the map stored state. And uses this id to pull the data view when the map is displayed. So we cannot use a dynamic id in this case.

Having said that, I decided not to implement this for the `ux` plugin either, it'll require a of refactoring, and this would be out of the scope of this issue.

We can probably think about doing it when we see the need to use the APM data view again on other plugins.

Reply:
Not sure how `maps` is used in the above case but in APM I was able to use the the ad-hoc data view to render the geo map for the mobile service. 

https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx#L134-L137 

>  it'll require a of refactoring, and this would be out of the scope of this issue.
> we can probably think about doing it when we see the need to use the APM data view again on other plugins.

I agree, let's make sure we have a ticket for it